certora/specs/DoubleEndedQueue.spec

```
@@ -0,0 +1,240 @@
+import "helpers.spec"
+
+methods {
+    pushFront(bytes32)                    envfree
+    pushBack(bytes32)                     envfree
+    popFront()          returns (bytes32) envfree
+    popBack()           returns (bytes32) envfree
+    clear()                               envfree
+
+    // exposed for FV
+    begin()             returns (int128)  envfree
+    end()               returns (int128)  envfree
+    
+    // view
+    length()            returns (uint256) envfree
+    empty()             returns (bool)    envfree
+    front()             returns (bytes32) envfree
+    back()              returns (bytes32) envfree
+    at_(uint256)        returns (bytes32) envfree // at is a reserved word
+}
+
+/*
+┌─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
+│ Helpers                                                                                                             │
+└─────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
+*/
+
+function absolute_diff(mathint a, mathint b) returns mathint {
+    return a > b ? a - b : b - a; 
+}
+
+function min_int128() returns mathint {
+    return max_uint256 / 2 + 1;
+}
+
+function max_int128() returns mathint {
+    return max_uint256 / 2;
```

This should be available
```suggestion
    return max_uint128;
```

---

Thanks for this one!